### ISSUE TITLE
hashlink generated c code for hxd.App has syntax errors

### ISSUE THREAD

**Issue by dmlary** [HeapsIO/heaps#1132]
## Summary
hashlink generated code for basic `hxd.App` has syntax errors.  Specifically multiple `sdl_*()` functions prefixed with a `?` in `hl/natives.h`.

```c
// Natives functions
HL_API vbyte* ?sdl_get_clipboard_text(void);
HL_API vdynamic* ?sdl_get_current_display_mode(int,bool);
HL_API varray* ?sdl_get_display_modes(int);
HL_API varray* ?sdl_get_displays(void);
HL_API int ?sdl_get_framerate(sdl_window*);
HL_API int ?sdl_get_screen_height(void);
HL_API int ?sdl_get_screen_height_of_window(sdl_window*);
HL_API int ?sdl_get_screen_width(void);
HL_API int ?sdl_get_screen_width_of_window(sdl_window*);
HL_API void ?sdl_gl_compressed_tex_sub_image2d(int,int,int,int,int,int,int,int,vbyte*);
HL_API void ?sdl_gl_compressed_tex_sub_image3d(int,int,int,int,int,int,int,int,int,int,vbyte*);
HL_API void ?sdl_gl_tex_sub_image2d(int,int,int,int,int,int,int,int,vbyte*);
HL_API void ?sdl_gl_tex_sub_image3d(int,int,int,int,int,int,int,int,int,int,vbyte*);
HL_API bool ?sdl_set_clipboard_text(vbyte*);
HL_API int ?sdl_win_display_handle(sdl_window*);
HL_API bool ?sdl_win_set_display_mode(sdl_window*,int,int,int);
```

## Details
Trying out heaps with native compilation on macOS, and hit a code generation problem.

### source code (src/Main.hx):
```hx
class Main extends hxd.App {
    override function init() {
        trace("Hello World");
    }
    static function main() {
        new Main();
    }
}
```

### build steps:
```sh
haxe -cp src -lib heaps -lib hlsdl -hl build/main.c -main Main -debug
clang -o hello -Ibuild/ -I/usr/local/Cellar/hashlink/1.13/libexec/include build/main.c /usr/local/Cellar/hashlink/1.13/libexec/lib/libhl.dylib
```

### Compiler warnings:
```txt
In file included from build/main.c:10:
In file included from build/hl/functions.c:37:
build/hl/natives.h:32:15: error: expected identifier or '('
HL_API vbyte* ?sdl_get_clipboard_text(void);
              ^
build/hl/natives.h:33:18: error: expected identifier or '('
HL_API vdynamic* ?sdl_get_current_display_mode(int,bool);
                 ^
build/hl/natives.h:34:16: error: expected identifier or '('
HL_API varray* ?sdl_get_display_modes(int);
               ^
build/hl/natives.h:35:16: error: expected identifier or '('
HL_API varray* ?sdl_get_displays(void);
               ^
build/hl/natives.h:36:12: error: expected identifier or '('
HL_API int ?sdl_get_framerate(sdl_window*);
           ^
build/hl/natives.h:37:12: error: expected identifier or '('
HL_API int ?sdl_get_screen_height(void);
           ^
build/hl/natives.h:38:12: error: expected identifier or '('
HL_API int ?sdl_get_screen_height_of_window(sdl_window*);
           ^
build/hl/natives.h:39:12: error: expected identifier or '('
HL_API int ?sdl_get_screen_width(void);
           ^
build/hl/natives.h:40:12: error: expected identifier or '('
HL_API int ?sdl_get_screen_width_of_window(sdl_window*);
           ^
build/hl/natives.h:41:13: error: expected identifier or '('
HL_API void ?sdl_gl_compressed_tex_sub_image2d(int,int,int,int,int,int,int,int,vbyte*);
            ^
build/hl/natives.h:42:13: error: expected identifier or '('
HL_API void ?sdl_gl_compressed_tex_sub_image3d(int,int,int,int,int,int,int,int,int,int,vbyte*);
            ^
build/hl/natives.h:43:13: error: expected identifier or '('
HL_API void ?sdl_gl_tex_sub_image2d(int,int,int,int,int,int,int,int,vbyte*);
            ^
build/hl/natives.h:44:13: error: expected identifier or '('
HL_API void ?sdl_gl_tex_sub_image3d(int,int,int,int,int,int,int,int,int,int,vbyte*);
            ^
build/hl/natives.h:45:13: error: expected identifier or '('
HL_API bool ?sdl_set_clipboard_text(vbyte*);
            ^
build/hl/natives.h:46:12: error: expected identifier or '('
HL_API int ?sdl_win_display_handle(sdl_window*);
           ^
build/hl/natives.h:47:13: error: expected identifier or '('
HL_API bool ?sdl_win_set_display_mode(sdl_window*,int,int,int);
            ^
In file included from build/main.c:10:
build/hl/functions.c:8272:84557: error: expected expression
build/hl/functions.c:8272:84622: error: expected expression
build/hl/functions.c:8272:84558: error: use of undeclared identifier 'sdl_gl_compressed_tex_sub_image3d'; did you mean 'sdl_gl_compressed_tex_image3d'?
build/hl/natives.h:236:13: note: 'sdl_gl_compressed_tex_image3d' declared here
HL_API void sdl_gl_compressed_tex_image3d(int,int,int,int,int,int,int,int,vbyte*);
            ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

### Version info:
```txt
> haxelib list
format: [3.5.0]
hashlink: [0.1.0]
heaps: [1.9.1]
hlopenal: [1.5.0]
hlsdl: [1.13.0]
> haxe --version
4.2.5
> hl --version
1.13.0%
> clang --version
Homebrew clang version 15.0.7
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /usr/local/opt/llvm/bin
```

**Comment by ncannasse** [comment]
This has been fixed in Haxe compiler. You can download a build from haxe.org

Le dim. 19 févr. 2023 à 04:02, David M. Lary ***@***.***> a
écrit :

**Comment by dmlary** [comment]
That’s odd, I’m already running haxe 4.2.5, which is the latest version listed on the website.  Is there another way (besides `haxe —version` I can confirm the fix should be present?

**Comment by ncannasse** [comment]
It's not yet been released, the fix will be part of haxe 4.3.0, you can download nightly builds here : https://build.haxe.org/builds/haxe/
